### PR TITLE
Adds new OD cal and pauses stirring

### PR DESCRIPTION
**Pull request by odcambc**
This is a set of changes to OD calibrations that make them more reliable and simplify parallel calibrations, in case users have a set of standards.

The new action, `od_calibration_from_standards`, is adapted from `od_calibration`. Rather than a series of dilutions from a starting OD, however, it prompts a series of measurements and associated offline OD values, followed by a final media blank. It performs the fit as with `od_calibration`.

This also includes a change to `od_calibration` behavior. I found that the stir bar would intermittently fail to catch while replacing the vial during calibration - it pauses now and restarts before the next OD measurements. This should improve reliability of measurements. This required passing the Stirrer explicitly, which I'm not sure is the best pattern, but seemed the simplest way.

Tested on a local pioreactor - calibrated successfully.

Alternatively, should there be an option to change RPM during calibration? Similar to how PWM/hz are prompted during pump calibration.

### CODE REVIEW

**Comment by CamDavidsonPilon** [comment]
Hi @odcambc, this is exciting! 

 - In the PR, can you provide a high-level outline of how this calibration is performed?
 - How you tried running this locally?

What are your thoughts on making this a Pioreactor plugin? I don't yet know the line between built-in vs plugins, or if there will be a process to vendor popular plugins into the app. Just curious what your thoughts are.

File: pioreactor/actions/od_calibration_from_standards.py [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+            sleep(0.5)
+
+        click.echo(".", nl=False)
+        voltage = get_voltage_from_adc()
+        click.echo(".", nl=False)
+
+        od600_values.append(standard_od)
+        voltages.append(voltage)
+
+        st.set_target_rpm(0)
```
Comment (CamDavidsonPilon) [inline pioreactor/actions/od_calibration_from_standards.py:178]: An option that may be more responsive: `st.set_state("sleeping")` to pause stirring, and `st.set_state("ready")` to restart stirring at the correct duty cycle.

Comment (odcambc) [reply pioreactor/actions/od_calibration_from_standards.py]: This is better, thanks - this should keep block_until_rpm_is_close_to_target still, though, right? on_sleeping_to_ready doesn't block, I think...

**Comment by odcambc** [comment]
Made the PR a little more detailed! Sorry.

> What are your thoughts on making this a Pioreactor plugin? I don't yet know the line between built-in vs plugins, or if there will be a process to vendor popular plugins into the app. Just curious what your thoughts are.

Sure, the new action makes sense as a plugin, being self-contained, and most useful to users with multiple vials and units.  I can try to refactor that out and turn this into a plugin!

File: pioreactor/actions/od_calibration_from_standards.py (hunk 1 above)
Comment (CamDavidsonPilon) [reply pioreactor/actions/od_calibration_from_standards.py]: I think keeping `block_until_rpm_is_close_to_target` is a good idea, yea. `on_sleeping_to_ready` doesn't block

**Comment by odcambc** [comment]
Closing, split into plugin.

**Pull request closed without merging**